compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -74,57 +74,75 @@ object DesugarEnums {
     else if (isEnumCase(cdef)) cdef.withMods(cdef.mods.withFlags(cdef.mods.flags | Final))
     else cdef
 
+  private def valuesDotTerm(name: TermName)(implicit src: SourceFile) =
```

Comment:
It's shorter and more elegant to have a single `valuesDot` definition that takes a `PreName`:

```scala
private def valuesDotTerm(name: PreName)(implicit src: SourceFile) =
  Select(Ident(nme.DOLLAR_VALUES), name.toTermName)
```